crates/ntx-builder/src/builder/data_store/mod.rs

```diff
@@ -0,0 +1,142 @@
+use std::{collections::BTreeSet, sync::Arc};
+
+use account_cache::AccountCache;
+use miden_objects::{
+    AccountError, MastForest, Word,
+    account::{Account, AccountId},
+    block::{BlockHeader, BlockNumber},
+    crypto::merkle::PartialMmr,
+    note::{NoteScript, NoteTag},
+    transaction::{ExecutedTransaction, PartialBlockchain},
+};
+use miden_tx::{DataStore, DataStoreError, MastForestStore, TransactionMastStore};
+use tokio::sync::Mutex;
+use tracing::warn;
+
+use super::store::{StoreClient, StoreError};
+use crate::COMPONENT;
+
+mod account_cache;
+
+// DATA STORE
+// =================================================================================================
+
+pub struct NtxBuilderDataStore {
+    mast_forest_store: TransactionMastStore,
+    store_client: StoreClient,
+    account_cache: AccountCache,
+    block_ref: Mutex<BlockHeader>,
+    partial_mmr: Mutex<PartialMmr>,
+}
+
+impl NtxBuilderDataStore {
+    pub async fn new(store_client: StoreClient) -> Result<Self, StoreError> {
+        let account_cache = AccountCache::new(128);
+        let mast_forest_store = TransactionMastStore::new();
+
+        // SAFETY: ok to unwrap because passing `None` should return the latest data everytime
```

Comment:
nit: how do we feel about using `expect()` instead of safety comments for unwraps?